tests/claim.zil.test.ts

```diff
@@ -305,6 +332,32 @@ describe("staking contract", () => {
             await increaseBNum(zilliqa, 10);
           },
           error: undefined,
+          want: {
+            events: [
+              {
+                name: "check_rewards",
+                getParams: () => ({
+                  rewards: ["List (Pair (ByStr20) (Uint128))",
+                  [
+                    [globalToken1ContractAddress,"10000000000000"],
+                    [globalToken2ContractAddress,"10000000000000"],
+                  ]
+                ]
+                }),
+              }
+            ],
+            verifyEvents: verifyEventsVerbose,
```

Comment:
Can we use the specific solution for this case? This way the function is more readable and easier to modify.

```js
        verifyEvents: (events, want) => {
          for (const [index, event] of events.entries()) {
            expect(event._eventname).toBe(want[index].name);

            const wantParams = getJSONParams(want[index].getParams());
            event.params[0].value.sort((a, b) =>
              a.arguments[0] < b.arguments[0] ? 1 : -1
            );
            wantParams[0]?.value.sort((a, b) =>
              a.arguments[0] < b.arguments[0] ? 1 : -1
            );
            expect(JSON.stringify(event.params)).toBe(
              JSON.stringify(wantParams)
            );
          }
          return true;
        },
```

Reply:
This will cause error though:

![image](https://user-images.githubusercontent.com/8149051/145773814-72f2573e-76f3-4ac2-86dc-cf094926f25e.png)

Reply:
That's because the number of events is more then you expect.
check `events.length` and `want.length`

Reply:
but let me try simplify function `verifyEventsVerbose` using your ides

Reply:
sure

Reply:
I think `verifyEventsVerbose` can't check the number of expected events.

Reply:
when I check the `events.length` and `want.length`, it was `4` and `1`. Now it might be `3` and `1`.

Reply:
These are the full events.
```
[
    {"_eventname":"check_rewards","address":"0xa744160c3de133495ab9f9d77ea54b325b045670","params":[{"type":"List (Pair (ByStr20) (Uint128))","value":[{"argtypes":["ByStr20","Uint128"],"arguments":["0x1e23eab4a9b56104f8e3bc364aff5b9fcc37d473","10000000000000"],"constructor":"Pair"},{"argtypes":["ByStr20","Uint128"],"arguments":["0x0c23b9e61e5fe6d9810543dc5fb9dfb7f0019549","10000000000000"],"constructor":"Pair"}],"vname":"rewards"}]},
    {"_eventname":"setup_rewards_distribution","address":"0xa744160c3de133495ab9f9d77ea54b325b045670","params":[{"type":"ByStr20","value":"0x0c23b9e61e5fe6d9810543dc5fb9dfb7f0019549","vname":"token_address"},{"type":"Uint128","value":"10000000000000","vname":"reward"}]},
    {"_eventname":"setup_rewards_distribution","address":"0xa744160c3de133495ab9f9d77ea54b325b045670","params":[{"type":"ByStr20","value":"0x1e23eab4a9b56104f8e3bc364aff5b9fcc37d473","vname":"token_address"},{"type":"Uint128","value":"10000000000000","vname":"reward"}]},
    {"_eventname":"calculate_rewards_tmp_inner","address":"0xa744160c3de133495ab9f9d77ea54b325b045670","params":[{"type":"Uint32","value":"1","vname":"cycle"},{"type":"Uint128","value":"10","vname":"total_stake"},{"type":"Uint128","value":"10","vname":"user_stake_this_cycle"}]}
]
```

Reply:
Good, one more thing `verifyEventsVerbose` can we rename `verifyCheckRewardsEvents`?

Reply:
make sense